Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/D7AbsoluteCrossSections.py

```diff
@@ -514,71 +549,95 @@ def _set_units(self, ws, nMeasurements):
+        cs_present = not self.getProperty("CrossSectionSeparationMethod").isDefault
+        possible_polarisations = ["ZPO", "YPO", "XPO", "XMYPO", "XPYPO"]  # 10p is approximate, no data atm
+        flipper_states = ["_ON", "_OFF"]
+        merging_keys = dict()
         for name in mtd[ws].getNames():
-            last_underscore = name.rfind("_")
-            if name[last_underscore+1:] == 'normalised':
-                short_name = name[:last_underscore]
-                second_last = short_name.rfind("_")
-                numors.add(short_name[:second_last])
-                pol_directions.add(short_name[second_last + 1:]+name[last_underscore:])
+            if cs_present:
+                non_numor_info = name[name.find("_")+1:]
+                next_underscore = non_numor_info.find("_")
+                if next_underscore < 1:
+                    key = non_numor_info
+                else:
+                    key = non_numor_info[:next_underscore]
             else:
-                numors.add(name[:last_underscore])
-                pol_directions.add(name[last_underscore + 1:])
-        if len(numors) > 1:
-            names_list = []
-            for direction in sorted(list(pol_directions)):
-                name = '{0}_{1}'.format(ws, direction)
-                list_pol = []
-                for numor in numors:
-                    list_pol.append('{0}_{1}'.format(numor, direction))
-                self._call_sum_data(input_name=','.join(list_pol), output_name=name)
+                key = ""
+                for pol in possible_polarisations:
+                    if pol in name:
+                        key = pol
+                        break
+                if flipper_states[0] in name:
+                    key += flipper_states[0]
+                else:
+                    key += flipper_states[1]
+            if key not in merging_keys:
+                merging_keys[key] = list()
+            merging_keys[key].append(name)
+        if len(merging_keys) > 1:
+            names_list = list()
+            for key in merging_keys:
+                name = '{0}_{1}'.format(ws, key)
+                self._call_sum_data(input_name=merging_keys[key], output_name=name)
                 names_list.append(name)
             DeleteWorkspaces(WorkspaceList=ws)
             GroupWorkspaces(InputWorkspaces=names_list, OutputWorkspace=ws)
         return ws
 
     def _call_sum_data(self, input_name, output_name=''):
+        """Wrapper around SumOverlappingTubes algorithm."""
         if output_name == '':
             output_name = input_name
         SumOverlappingTubes(InputWorkspaces=input_name, OutputWorkspace=output_name,
                             OutputType='1D', ScatteringAngleBinning=self.getProperty('ScatteringAngleBinSize').value,
                             Normalise=True, HeightAxis='-0.1,0.1')
         return output_name
 
+    def _set_output_names(self, output_ws):
```

Comment:
Another missing docstring, even though the name says it all, but it's pleasant to see consistency everywhere.